modules/disk-uploader/pkg/vmexport/vmexport_test.go

```diff
@@ -0,0 +1,263 @@
+
+		kubecli.GetKubevirtClientFromClientConfig = kubecli.GetMockKubevirtClientFromClientConfig
+		kubecli.MockKubevirtClientInstance = kubecli.NewMockKubevirtClient(ctrl)
+		kubecli.MockKubevirtClientInstance.EXPECT().CoreV1().Return(kubeClient.CoreV1()).AnyTimes()
+		kubecli.MockKubevirtClientInstance.EXPECT().VirtualMachineExport(namespace).Return(vmExportClient.ExportV1beta1().VirtualMachineExports(namespace)).AnyTimes()
+
+		virtClient, _ = kubecli.GetKubevirtClientFromClientConfig(nil)
+	})
+
+	Describe("CreateVirtualMachineExport", func() {
+		Context("when the VirtualMachineExport created successfully", func() {
+			BeforeEach(func() {
+				kubeClient.Fake.PrependReactor("get", "pods", func(action testing.Action) (bool, runtime.Object, error) {
+					_, ok := action.(testing.GetAction)
+					Expect(ok).To(BeTrue())
+
+					pod := &corev1.Pod{
+						ObjectMeta: metav1.ObjectMeta{
+							Name:      name,
+							Namespace: namespace,
+						},
+					}
+					return true, pod, nil
+				})
+				vmExportClient.Fake.PrependReactor("create", "virtualmachineexports", func(action testing.Action) (bool, runtime.Object, error) {
+					create, ok := action.(testing.CreateAction)
+					Expect(ok).To(BeTrue())
+
+					vmExport, ok := create.GetObject().(*v1beta1.VirtualMachineExport)
+					Expect(ok).To(BeTrue())
+					return true, vmExport, nil
+				})
+
+				os.Setenv("POD_NAME", name)
+				os.Setenv("POD_NAMESPACE", namespace)
+			})
+
+			AfterEach(func() {
+				os.Unsetenv("POD_NAME")
+				os.Unsetenv("POD_NAMESPACE")
+			})
+
+			DescribeTable("should return no error when created",
+				func(resource string) {
+					err := vmexport.CreateVirtualMachineExport(kubeClient, virtClient, resource, namespace, name)
+					Expect(err).NotTo(HaveOccurred())
+				},
+				Entry("export-source-kind is vm", "vm"),
+				Entry("export-source-kind is vmsnapshot", "vmsnapshot"),
+				Entry("export-source-kind is pvc", "pvc"),
+			)
+		})
+
+		Context("when the VirtualMachineExport creation fails", func() {
+			BeforeEach(func() {
+				os.Setenv("POD_NAME", name)
+				os.Setenv("POD_NAMESPACE", namespace)
+			})
+
+			AfterEach(func() {
+				os.Unsetenv("POD_NAME")
+				os.Unsetenv("POD_NAMESPACE")
+			})
+
+			It("should return error when export-source-kind invalid", func() {
+				err := vmexport.CreateVirtualMachineExport(kubeClient, virtClient, "fake", namespace, name)
+				Expect(err).To(HaveOccurred())
+				Expect(err.Error()).To(ContainSubstring("invalid export-source-kind: fake, must be one of vm, vmsnapshot, pvc"))
+			})
+
+			It("should return error when set pod owner reference failed", func() {
+				err := vmexport.CreateVirtualMachineExport(kubeClient, virtClient, "vm", namespace, name)
+				Expect(err).To(HaveOccurred())
+				Expect(err.Error()).To(ContainSubstring("pods \"%s\" not found", name))
+			})
+		})
+	})
+
+	Describe("WaitUntilVirtualMachineExportReady", func() {
+		Context("when the VirtualMachineExport is ready", func() {
```

Comment:
These `Describe` and `Context()` contain only a single test. Can you remove them and move the `It()` into this level?

Reply:
Yep!